spec/transaction_repository_spec.rb

```diff
@@ -0,0 +1,243 @@
+      allow_any_instance_of(TransactionRepository).to receive(:create_transactions).and_return(mock_data)
+      t_repo = TransactionRepository.new('fake.csv')
+
+      attributes = {
+        id: nil,
+        invoice_id: 8,
+        credit_card_number: '1212121212121212',
+        credit_card_expiration_date: '0222',
+        result: 'success',
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      new_transactions = t_repo.create(attributes)
+
+      expect(new_transactions.last).is_a? Transaction
+      expect(new_transactions.last.id).to eq 7
+      expect(new_transactions.length).to eq 11
+    end
+  end
+
+  describe '#update' do
+    it 'updates the correct Transaction and attributes' do
+      mock_data = MockData.mock_generator(self, 'Transaction', data_hashes)
+      allow_any_instance_of(TransactionRepository).to receive(:create_transactions).and_return(mock_data)
+      t_repo = TransactionRepository.new('fake.csv')
+
+      attributes = {
+        id: nil,
+        invoice_id: 8,
+        credit_card_number: '1212121212121212',
+        credit_card_expiration_date: '0222',
+        result: 'failure',
+        created_at: Time.now,
+        updated_at: Time.now
+      }
+      new_transactions = t_repo.create(attributes)
+      new_info = {
+        credit_card_number: '1212121212124444',
+        credit_card_expiration_date: '0224',
+        result: 'success'
+      }
+      t_repo.update(7, new_info)
+      updated_item = t_repo.find_by_id(7)
+
+      expect(updated_item.credit_card_number).to eq '1212121212124444'
+      expect(updated_item.credit_card_expiration_date).to eq '0224'
+      expect(updated_item.result).to eq 'success'
+    end
+
+    it 'does not update immutable attributes' do
+      mock_data = MockData.mock_generator(self, 'Transaction', data_hashes)
+      allow_any_instance_of(TransactionRepository).to receive(:create_transactions).and_return(mock_data)
+      t_repo = TransactionRepository.new('fake.csv')
+
+      attributes = {
+        id: nil,
+        invoice_id: 8,
+        credit_card_number: '1212121212121212',
+        credit_card_expiration_date: '0222',
+        result: 'failure',
+        created_at: Time.new(2021, 0o3, 0o1),
+        updated_at: Time.new(2021, 0o3, 0o1)
+      }
+      new_transactions = t_repo.create(attributes)
+      new_info = {
+        id: 10,
+        invoice_id: 12,
+        credit_card_number: '1212121212124444',
+        credit_card_expiration_date: '0224',
+        result: 'success',
+        created_at: Time.new(2021, 0o2, 0o4),
+        updated_at: Time.new(2021, 0o2, 0o4)
+      }
+      t_repo.update(7, new_info)
+      updated_item = t_repo.transactions.last
+
+      expect(updated_item.id).to eq 7
+      expect(updated_item.invoice_id).to eq 8
+      expect(updated_item.created_at).to eq Time.new(2021, 0o3, 0o1)
+      expect(updated_item.updated_at).to be > Time.new(2021, 0o3, 0o1)
```

Comment:
I like this Time test, I'm about to lift it from the spec harness for my repo class